src/frontend/mmdf.jl

```diff
@@ -56,93 +57,72 @@ Following arguments are set optionally:
     concentration_ratios = Dict{String,Tuple{String,String,Float64}}(),
     constant_concentrations = Dict{String,Float64}(),
     ignored_metabolites = [],
     proton_metabolites = [],
     water_metabolites = [],
-    concentration_lower_bound = 1e-9, # M
-    concentration_upper_bound = 1e-1, # M
+    concentration_lower_bound = 1e-9, # Molar
+    concentration_upper_bound = 1e-1, # Molar
     T = 298.15, # Kelvin
     R = 8.31446261815324e-3, # kJ/K/mol
     reference_flux_atol = 1e-6,
-    check_ignored_reactions = missing,
     settings = [],
     optimizer,
 )
 
+    #=
+    For MMDF, one is almost always interested in running the analysis on a
+    subset of the full model, because thermodynamic data is usually not
+    available for all the reactions/metabolites. Missing data may cause subtle
+    problems, and is usually best to restrict your model to reactions where
+    thermodynamic data exists.  
+    =#
+
+    reaction_subset =
+        isempty(reference_flux) ? A.reactions(model) :
+        collect(k for (k, v) in reference_flux if abs(v) > reference_flux_atol)
+
+    metabolite_subset =
+        isempty(reference_flux) ? A.metabolites(model) :
+        unique(
+            mid for rid in reaction_subset for
+            mid in keys(A.reaction_stoichiometry(model, string(rid)))
+        )
+
+
     # First let's just check if all the identifiers are okay because we use
     # quite a lot of these; the rest of the function may be a bit cleaner with
     # this checked properly.
 
-    model_reactions = Set(A.reactions(model))
-    model_metabolites = Set(A.metabolites(model))
-
-    all(in(model_reactions), keys(reaction_standard_gibbs_free_energies)) || throw(
+    all(in(Set(keys(reaction_standard_gibbs_free_energies))), reaction_subset) || throw(
         DomainError(
             reaction_standard_gibbs_free_energies,
-            "unknown reactions referenced by reaction_standard_gibbs_free_energies",
+            "some reactions are not accounted for in reaction_standard_gibbs_free_energies.",
         ),
     )
-    all(x -> haskey(reaction_standard_gibbs_free_energies, x), keys(reference_flux)) ||
-        throw(DomainError(reference_flux, "some reactions have no reference flux"))
-    all(in(model_reactions), keys(reference_flux)) || throw(
-        DomainError(
-            reaction_standard_gibbs_free_energies,
-            "unknown reactions referenced by reference_flux",
-        ),
+    all(in(Set(A.reactions(model))), reaction_subset) || throw(
+        DomainError(reference_flux, "unknown reactions referenced by reference_flux."),
     )
-    all(in(model_metabolites), keys(constant_concentrations)) || throw(
+    all(in(metabolite_subset), keys(constant_concentrations)) || throw(
         DomainError(
             constant_concentrations,
-            "unknown metabolites referenced by constant_concentrations",
+            "unknown metabolites referenced by constant_concentrations.",
```

Comment:
why is this suddenly a sentence? (either have a normal error message or convert everything to sentences consistently)